### PR TITLE
feat: remove borders and padding in editor to match front-end styles

### LINKED ISSUE #9: [design] Focused vs. unfocused styles
Screenshot 2020-11-06 at 10 50 23

When the Curated List block is unfocused, we shouldn't see the borders, paddings, [+], etc...

_Originally posted by @thomasguillot in https://github.com/Automattic/newspack-listings/pull/6#issuecomment-723015033_

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Removes borders and padding from block editor elements so that styles more closely match both core WP design patterns and the front-end styles for listings.

Closes #9.

### How to test the changes in this Pull Request:

1. Check out this branch and run `npm run build`.
2. Create or edit a curated listing block in Specific Listings mode and add several listings.
3. Observe that the styles for rendered listing items more closely match the front-end. Example:

In the editor:
Screen Shot 2020-12-04 at 10 19 16 AM

On the front-end:
Screen Shot 2020-12-04 at 10 22 04 AM

4. Test that it's still possible/easy enough to select the parent Curated List wrapper block in order to edit block attributes, both through the block navigation UI and by clicking near/alongside each listing item.
5. Repeat steps 2-4 with a list in Query mode.

#### Bonus: Background color option

This branch also adds a new option for setting the background color of a list. To test:

1. Create a new list in Specific Listings mode and add several listings.
2. Open the Color Settings panel and observe a new color picker for Background Color.
3. Confirm that the background color of the list changes in both the editor and front-end when you select a background color. Also confirm that a `1em` padding is added to the list container if there's a background color applied.
4. Repeat steps 1-3 with a list in Query mode.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/blocks/curated-list/editor.scss [hunk 1]
```diff
@@ -2,9 +2,10 @@
 
 .newspack-listings {
 	&__curated-list-editor {
-		border: 1px solid var( --newspack-listings--grey-dark );
 		border-radius: 2px;
 		clear: both;
+		margin-left: -1rem;
+		margin-right: -1rem;
 		padding: 1em;
```
[the remark this answers is not in the document]
Comment (dkoo) [reply src/blocks/curated-list/editor.scss]: Fixed in 0bc7fe9.

File: src/blocks/curated-list/editor.scss [hunk 2]
```diff
@@ -14,6 +15,14 @@
 			line-height: 1.5;
 			margin-bottom: 8px;
 		}
+
+		.wp-block[data-type^='newspack-listings']
+			+ .wp-block[data-type^='newspack-listings']
+			.newspack-listings__listing-editor,
+		.newspack-listings__listing-editor + .newspack-listings__listing-editor {
+			border-top: 1px solid var( --newspack-listings--grey-light );
```
[the remark this answers is not in the document]
Comment (dkoo) [reply src/blocks/curated-list/editor.scss]: This was a bit more complex than I anticipated, but I borrowed some tricks from the Campaigns plugin to determine whether the border color should be white vs. black, based on the contrast ratio with the background color. Updated in 0bc7fe9.

[the remark this answers is not in the document]
Comment (thomasguillot) [reply src/blocks/curated-list/editor.scss]: Oh wow yes I can see! It's great 🙂

**Review by thomasguillot — APPROVED** [review]
Thank you for this @dkoo! 🚢

**Pull request merged**